File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/location/replay/ReplayLocationDto.java
package com.mapbox.services.android.navigation.v5.location.replay;

import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;

import java.util.Date;

class ReplayLocationDto {

  @SerializedName("lng")
  private double longitude;
  @SerializedName("horizontalAccuracy")
  private float horizontalAccuracyMeters;
  @SerializedName("course")
  private double bearing;
  @SerializedName("verticalAccuracy")
  private float verticalAccuracyMeters;
  private double speed;
  @SerializedName("lat")
  private double latitude;
  @SerializedName("altitude")
  private double altitude;
  @SerializedName("timestamp")
  @JsonAdapter(TimestampAdapter.class)
  private Date date;

  double getLongitude() {
    return longitude;
  }

  void setLongitude(double longitude) {
    this.longitude = longitude;
  }

  float getHorizontalAccuracyMeters() {
    return horizontalAccuracyMeters;
  }

  void setHorizontalAccuracyMeters(float horizontalAccuracyMeters) {
    this.horizontalAccuracyMeters = horizontalAccuracyMeters;
  }

  double getBearing() {
    return bearing;
  }

  void setBearing(double bearing) {
    this.bearing = bearing;
  }

  float getVerticalAccuracyMeters() {
    return verticalAccuracyMeters;
  }

  void setVerticalAccuracyMeters(float verticalAccuracyMeters) {
    this.verticalAccuracyMeters = verticalAccuracyMeters;
  }

  double getSpeed() {
    return speed;
  }

  void setSpeed(double speed) {
    this.speed = speed;
  }

  double getLatitude() {
    return latitude;
  }

  void setLatitude(double latitude) {
    this.latitude = latitude;
  }

  double getAltitude() {
    return altitude;
  }

  void setAltitude(double altitude) {
    this.altitude = altitude;
  }

  Date getDate() {
    return date;
  }

  void setDate(Date date) {
    this.date = date;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/location/replay/ReplayLocationDto.kt
package com.mapbox.services.android.navigation.v5.location.replay

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName

import java.util.Date

class ReplayLocationDto {

    @SerializedName("lng")
    var longitude: Double = 0.toDouble()
    @SerializedName("horizontalAccuracy")
    var horizontalAccuracyMeters: Float = 0.toFloat()
    @SerializedName("course")
    var bearing: Double = 0.toDouble()
    @SerializedName("verticalAccuracy")
    var verticalAccuracyMeters: Float = 0.toFloat()
    var speed: Double = 0.toDouble()
    @SerializedName("lat")
    var latitude: Double = 0.toDouble()
    @SerializedName("altitude")
    var altitude: Double = 0.toDouble()
    @SerializedName("timestamp")
    @JsonAdapter(TimestampAdapter::class)
    var date: Date? = null
}
